docs/v3/core/index.rst

```diff
@@ -1598,38 +1508,162 @@ Storage transformers may be stacked to combine different functionalities:
-potential extensions.
-
-Specifications for new extensions are recommended to be published in the
-https://github.com/zarr-developers/zarr-specs repository via the
-`ZEP process <https://zarr.dev/zeps/active/ZEP0000.html>`_. If a specification
-is published decentralized (e.g. for initial experimentation or due to a very
-specialized scope), it must use a URL in the `name` key of its metadata, which
-identifies the publishing organization or individual, and should point to the
-specification of the extension.
-
-Future versions of this specification may also add new core features by adding new top-level
-metadata keys. Such features are required by default. However, if the value of an unknown feature
-is an object containing the key-value pair ``"must_understand": false``, it can be ignored.
+=========== ======================= ================================================================== ================================
+node_type   extension point         metadata definition                                                list of core extensions
+=========== ======================= ================================================================== ================================
+array       data type               :ref:`data-type <array-metadata-data-type>`                        :ref:`data-types-list`
+array       chunk grid              :ref:`chunk-grid <array-metadata-chunk-grid>`                      :ref:`chunk-grid-list`
+array       chunk key encoding      :ref:`chunk-key-encoding <array-metadata-chunk-key-encoding>`      :ref:`chunk-key-encoding-list`
+array       codecs                  :ref:`codecs <array-metadata-codecs>`                              :ref:`codecs-list`
+array       storage transformer     :ref:`storage-transformers <array-metadata-storage-transformers>`  :ref:`storage-transformers-list`
+=========== ======================= ================================================================== ================================
+
+New extension points may be proposed to the Zarr community through the ZEP
+process. See `ZEP 0 <https://zarr.dev/zeps/active/ZEP0000.html>`_ for more information.
+
+.. _extension-definition:
+
+Extension definition
+--------------------
+
+In `metadata documents`_, extensions can be encoded either as objects or as
+short-hand names. If using an objection definition, the member ``name``
+MUST be a plain string which conforms to :ref:`extension name <extension-naming>`.
+Optionally, the member ``configuration`` MAY be present but if so MUST be
+an object.
+
+For example::
+
+    {
+        "name": "<name>",        # "raw name" or URL-based name
+        "configuration": { ... } # optional
+    }
+
+Instead of extension objects, short-hand names MAY be used if no
+configuration metadata is required. They are equivalent to extension
+objects with just a `name` key.
+
+If such an object is present, the field `must_understand` is implicitly set to
```

Comment:
This part is referenced in different sections of this doc. Should it become its own (sub)section?